### ISSUE TITLE
Defer Loader fails to load raw headers from dev server. Works in production.

### ISSUE THREAD

**Issue by youngkidwarrior** [remix-run/remix#5489]
### What version of Remix are you using?

1.12.0

### Are all your remix dependencies & dev-dependencies using the same version?

- [X] Yes

### Steps to Reproduce

You could clone [BrightID Discord Bot](https://github.com/ShenaniganDApp/brightid-discord-bot/tree/assign-sp/apps%2Fweb)

### Expected Behavior

Loader returns deferred data properly

Here is a link to the loader (written in Rescript) https://github.com/ShenaniganDApp/brightid-discord-bot/blob/212d4dbdf1fa71682fb7064f923ada2a3a510edc/apps/web/app/res-routes/guilds/GuildId_Index.res#L27

### Actual Behavior

https://user-images.githubusercontent.com/36084652/219475522-83822cb4-dfb6-4a09-8a1e-02b4d13099cd.mov

I have a loader that is supposed to return deferred data. On page load, the loader is returning undefined for every field.

This is the error:
```
TypeError: Cannot read properties of undefined (reading 'raw')
    at sendRemixResponse (/Users/username/Documents/BrightIDbot/node_modules/@remix-run/express/dist/server.js:85:65)
    at /Users/username/Documents/BrightIDbot/node_modules/@remix-run/express/dist/server.js:40:13
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
```

and this is the function the error points to in `@remix-run/express`

```javascript
async function sendRemixResponse(res, nodeResponse) {
  res.statusMessage = nodeResponse.statusText;
  res.status(nodeResponse.status);
  for (let [key, values] of Object.entries(nodeResponse.headers.raw())) {  // --> Line 85
    for (let value of values) {
      res.append(key, value);
    }
  }
  if (nodeResponse.body) {
    await node.writeReadableStreamToWritable(nodeResponse.body, res);
  } else {
    res.end();
  }
}
```

The odd thing is, the exact same code works in production at https://bot.brightid.org, which makes me think it is a dev server issue. I recently updated to `v1.12.0`, so I tried reverting those changes, however, the issue persisted.

**Comment by MaximeGratens** [comment]
Same issue with my project !

**Comment by youngkidwarrior** [comment]
It seems like this issue is arising from `defer()`. Not sure how to fix it.

**Comment by bluefire2121** [comment]
This is fixed in 1.14.0 I believe

**Comment by robase** [comment]
Still hitting this issue using 1.14.3

**Comment by brophdawg11** [comment]
I'm going to close this out since I believe it should have been fixed by #5345 in `1.14.0`.  Please re-open with a minimal reproduction if this is not the case!
